test/models/repository.test.js

```diff
@@ -2125,4 +2124,193 @@ describe('Repository', function() {
       });
     });
   });
+
+  describe('updating commit message', function() {
+    let workdir, sub;
+    let observedEvents, eventCallback;
+
+    async function wireUpObserver(fixtureName = 'multi-commits-files', existingWorkdir = null) {
+      observedEvents = [];
+      eventCallback = () => {};
+
+      workdir = existingWorkdir || await cloneRepository(fixtureName);
+      const repository = new Repository(workdir);
+      await repository.getLoadPromise();
+
+      const observer = new FileSystemChangeObserver(repository);
+
+      sub = new CompositeDisposable(
+        new Disposable(async () => {
+          await observer.destroy();
+          repository.destroy();
+        }),
+      );
+
+      sub.add(observer.onDidChange(events => {
+        observedEvents.push(...events);
+        eventCallback();
+      }));
+
+      return {repository, observer};
+    }
+
+    function expectEvents(repository, ...suffixes) {
```

Comment:
Yeah, these would be great to extract to a helper and clean up a bit... We have a few places that test Real Filesystem Events :tm: where these would be nice.

Reply:
How does this look b3c9ec7?